app/assets/javascripts/components/shipment_version.js.coffee

```diff
@@ -4,47 +4,67 @@ window.ShipmentVersion = class ShipmentVersion extends React.Component
     super(props, context)
     @state = {
       shipment: props.shipment
+      index: props.index
       changes: props.changes
       rowType: props.rowType
+      userType: props.userType
     }
 
   render: ->
     data = @state.shipment
     keys = Object.keys(@state.changes)
-    changes = if keys.length > 1 then ' changes' else ' change'
+    changes = ''
+    if keys.length == 0
+      changes = 'deleted'
+    else if keys.length > 1
+      changes = "#{keys.length} changes"
+    else
+      changes = "1 change"
     tr({ className: @state.rowType },
       td({}
-        div({ className: 'bold' }, data.updated_at)
-        div({ className: 'italic' }, keys.length + changes)
+        div({ className: "bold" }, data.updated_at)
+        div({ className: "italic" }, changes)
       )
-      td({}, span({ className: 'appendix' }, data.appendix))
+      td({}, span({ className: "appendix" }, data.appendix))
       td({},
         div(
-          { className: 'taxon_name bold' }
+          { className: "#{@state.index}_taxon_name bold" }
           data.taxon_name
         )
-        div({ className: 'accepted-name' }, data.taxon_name) # Replace with accepted_name
+        div({ className: "#{@state.index}_accepted_name" }, data.taxon_name) # Replace with accepted_name
       )
-      td({}, span({ className: 'term' }, data.term))
-      td({}, span({ className: 'quantity' }, data.quantity))
-      td({}, span({ className: 'trading_partner' }, data.trading_partner))
-      td({}, span({ className: 'country_of_origin' }, data.country_of_origin))
-      td({}, span({ className: 'import_permit' }, data.import_permit))
-      td({}, span({ className: 'export_permit' }, data.export_permit))
-      td({}, span({ className: 'origin_permit' }, data.origin_permit))
+      td({}, span({ className: "#{@state.index}_term" }, data.term))
+      td({}, span({ className: "#{@state.index}_quantity" }, data.quantity))
+      td({}, span({ className: "#{@state.index}_trading_partner" }, data.trading_partner))
+      td({}, span({ className: "#{@state.index}_country_of_origin" }, data.country_of_origin))
+      td({}, span({ className: "#{@state.index}_import_permit" }, data.import_permit))
+      td({}, span({ className: "#{@state.index}_export_permit" }, data.export_permit))
+      td({}, span({ className: "#{@state.index}_origin_permit" }, data.origin_permit))
       td({}
-        span({ className: 'purpose_code'}, data.purpose_code)
-        ' - '
-        span({ className: 'source_code' }, data.source_code)
-        ' - '
-        span({ className: 'year' }, + data.year)
+        span({ className: "#{@state.index}_purpose_code"}, data.purpose_code)
+        " - "
+        span({ className: "#{@state.index}_source_code" }, data.source_code)
+        " - "
+        span({ className: "#{@state.index}_year" }, + data.year)
       )
+      td({}, span({ className: "#{@state.index}_updated_at"}, data.updated_at))
+      @showEditor()
     )
 
+  showEditor: ->
```

Comment:
does this apply to both shipments and versions? In that case we're not using the value of whodunnit?